code/UI/Inventory/InventoryPanel.razor

```diff
@@ -0,0 +1,73 @@
+@using Sandbox;
+
+@namespace Grubs
+
+@if (Game.LocalPawn is not Player player)
+{
+    return;
+}
+
+<root class="@(_isOpen ? "open" : "")">
+    <label class="name" text=@_hoveredWeapon?.Name />
+    <div class="items">
+        @foreach(var weapon in player.Inventory.Weapons)
+        {
+            var isAvailable = weapon.Charges > 0;
+            <InventoryItem
+             Weapon=@weapon
+             IsAvailable=@isAvailable
+             @onmouseover=@(() => _hoveredWeapon = weapon)
+             @onmouseout=@(() => _hoveredWeapon = null)
+             @onclick=@(() => { if (isAvailable) player.Inventory.SetActiveWeapon(weapon); })
```

Comment:
SetActiveWeapon needs to be changed, it doesn't work when called from client. I'd rather do it in another PR.

1. Either we change the weapon on server (i.e make cmd call)
2. Or we setup `[ClientInput] ActiveChildInput` (this would be preferable

Reply:
We can set up client input for that. If you want to have a go at it this PR feel free. This all looks great, btw.